bser/encode.go

```diff
@@ -150,19 +150,21 @@ func encode(buf []byte, d interface{}) ([]byte, error) {
 			}
 			return appendItem(buf, 0x00, b), nil
 		case reflect.Struct:
-			num := r.NumField()
-			b, err := encode(nil, num)
+			exportedFields, err := exportedStructFields(r.Type())
+			if err != nil {
+				return nil, err
+			}
+			b, err := encode(nil, len(exportedFields))
 			if err != nil {
 				return nil, err
 			}
 
-			for i := 0; i < num; i++ {
-				fieldName := r.Type().Field(i).Name
-				b, err = encode(b, fieldName)
+			for _, field := range exportedFields {
+				b, err = encode(b, field.Name)
 				if err != nil {
 					return nil, err
 				}
-				fv := r.Field(i)
+				fv := r.FieldByName(field.Name)
```

Comment:
probably better to use FieldByIndex here, the implementation looks more efficient than FieldByName in https://golang.org/src/reflect/type.go

Reply:
@jonasi good call, updated